includes/classes/admin/editor/class-shortcodes.php

```diff
@@ -133,13 +133,127 @@ public function pip_breadcrumb() {
          * @return string
          */
         public function pip_title() {
-            // If AJAX, display default message
-            if ( wp_doing_ajax() ) {
-                return __( 'Title here', 'pilopress' );
+
+            $title = '';
+
+            //* Front
+            if ( is_singular() ) {
+
+                //* Post / Page...
+                $title = get_the_title();
+
+            } else {
+
+                //* Taxonomy / Category / Tag / Archive / User...
+                $title = get_the_archive_title();
+
             }
 
-            // Render shortcode
-            return get_the_title();
+            //* Admin
+            if ( is_admin() ) {
+
+                //* Admin screen data
+                $screen_obj  = get_current_screen();
+                $screen_data = (array) $screen_obj;
+
+                //* Admin - GET data (WordPress)
+                $screen_base = acf_maybe_get( $screen_data, 'base' );
+
+                //* Compatibility fix for ACFE archive
+                if ( stripos( $screen_base, '-archive' ) !== false ) {
+                    $screen_base = 'archive';
+                }
+
+                switch ( $screen_base ) {
+
+                    //* Post type
+                    case 'post':
+                        $found_id = acf_maybe_get_GET( 'post' );
+                        $title    = get_the_title( $found_id );
+                        break;
+
+                    //* Term
+                    case 'term':
+                        $found_id         = acf_maybe_get_GET( 'tag_ID' );
+                        $found_term_obj   = get_term( $found_id );
+                        $found_term_data  = (array) $found_term_obj;
+                        $found_term_title = acf_maybe_get( $found_term_data, 'name' );
+                        $title            = $found_term_title ? $found_term_title : $title;
+                        break;
+
+                    //* ACFE Archive (option)
+                    case 'archive':
+                        $found_post_type  = acf_maybe_get_GET( 'post_type' );
+                        $post_type_obj    = get_post_type_object( $found_post_type );
+                        $post_type_data   = (array) $post_type_obj;
+                        $post_type_labels = acf_maybe_get( $post_type_data, 'labels' );
+                        $post_type_labels = (array) $post_type_labels;
+                        $post_type_label  = acf_maybe_get( $post_type_labels, 'archives' ) ? acf_maybe_get( $post_type_labels, 'archives' ) : acf_maybe_get( $post_type_data, 'label' );
+                        $title            = $post_type_label ? $post_type_label : $title;
```

Comment:
really not important but `$post_type_label ?: $title;` ? (other occurences in the file)

Also the `*` in these `//*` is just for style no ?

Reply:
@barnemax-pilotin  
- Kind of for style yeah and better readability (because i have an vscode extension that highlight comments based on the 1st symbol, cf screen) but i can remove it, it's ok.
![image](https://user-images.githubusercontent.com/62112531/103629178-4b9cca80-4f40-11eb-91c5-1ab997c7b747.png)

- Since we aim for specific PHP versions, i delibarately used the oldest syntax for compatibility.

Reply:
For consistency in code, I remove the `*` in you `//*`